Sources/protoc-gen-swift/MessageGenerator.swift

```diff
@@ -13,855 +13,550 @@
-        p.indent()
-        oneofHandled.removeAll(keepingCapacity: true)
-        for f in fields {
-            if let o = f.oneof {
-                if !oneofHandled.contains(f.descriptor.oneofIndex) {
-                    p.print("if \(o.swiftStorageFieldName) != other.\(o.swiftStorageFieldName) {return false}\n")
-                    oneofHandled.insert(f.descriptor.oneofIndex)
-                }
-            } else {
-                let notEqualClause = f.generateNotEqual(name: f.swiftStorageName, usesHeapStorage: true)
-                p.print("if \(notEqualClause) {return false}\n")
-            }
-        }
-        if !isProto3 {
-            p.print("if unknownFields != other.unknownFields {return false}\n")
-        }
-        if isExtensible {
-            p.print("if extensionFieldValues != other.extensionFieldValues {return false}\n")
-        }
-        p.print("return true\n")
-        p.outdent()
-        p.print("}\n")
+    for o in oneofs {
+      o.generateNested(printer: &p)
+    }
 
-        // copy helper
-        p.print("\n")
-        p.print("func copy() -> _StorageClass {\n")
-        p.indent()
-        p.print("let clone = _StorageClass()\n")
-        if !isProto3 {
-            p.print("clone.unknownFields = unknownFields\n")
-        }
-        if isExtensible {
-            p.print("clone.extensionFieldValues = extensionFieldValues\n")
-        }
-        oneofHandled.removeAll(keepingCapacity: true)
-        for f in fields {
-            if let o = f.oneof {
-                if !oneofHandled.contains(f.descriptor.oneofIndex) {
-                    p.print("clone.\(o.swiftStorageFieldName) = \(o.swiftStorageFieldName)\n")
-                    oneofHandled.insert(f.descriptor.oneofIndex)
-                }
-            } else {
-                p.print("clone.\(f.swiftStorageName) = \(f.swiftStorageName)\n")
-            }
-        }
-        p.print("return clone\n")
-        p.outdent()
-        p.print("}\n")
+    // Nested enums
+    for e in enums {
+      e.generateNested(printer: &p)
+    }
 
-        p.outdent()
-        p.print("}\n")
+    // Nested messages
+    for m in messages {
+      m.generateNested(printer: &p, file: file, parent: self)
     }
 
-    func generateIsInitialized(printer p: inout CodePrinter) {
-      var functionStarted = false;
-      func ensureFunctionStarted() {
-        if functionStarted { return }
-        functionStarted = true
-        p.print("\nvar isInitialized: Bool {\n")
-        p.indent()
+    // Nested extension declarations
+    if !extensions.isEmpty {
+      p.print("\n")
+      p.print("struct Extensions {\n")
+      p.indent()
+      for e in extensions {
+          e.generateNested(printer: &p)
       }
+      p.outdent()
+      p.print("}\n")
+    }
 
-      if isExtensible {
-        ensureFunctionStarted()
-        p.print("if !extensionFieldValues.isInitialized {return false}\n")
+    // ivars
+    if storage != nil {
```

Comment:
You could move this block up and combine it with the case that generates the storage class.  That way, the ivars (in the else clause below) would be in about the same place in the file regardless.